kernel/src/transaction.rs

```diff
@@ -310,77 +302,55 @@ pub enum CommitResult {
-            engine_commit_info.len()
-        )));
-    }
-
-    let commit_info_exprs = [
-        Expression::literal(timestamp),
-        Expression::literal(operation.unwrap_or(UNKNOWN_OPERATION)),
-        // HACK (part 1/2): since we don't have proper map support, we create a literal struct with
-        // one null field to create data that serializes as "operationParameters": {}
-        Expression::literal(Scalar::Struct(StructData::try_new(
-            vec![StructField::nullable(
-                "operation_parameter_int",
-                DataType::INTEGER,
-            )],
-            vec![Scalar::Null(DataType::INTEGER)],
-        )?)),
-        Expression::literal(format!("v{KERNEL_VERSION}")),
-        column_expr!("engineCommitInfo"),
-    ];
-    let commit_info_expr = Expression::struct_from([Expression::struct_from(commit_info_exprs)]);
-    let commit_info_schema = get_log_commit_info_schema().as_ref();
+    let mut commit_info_schema = get_log_commit_info_schema().as_ref().clone();
 
-    // HACK (part 2/2): we need to modify the commit info schema to match the expression above (a
-    // struct with a single null int field).
-    let mut commit_info_empty_struct_schema = commit_info_schema.clone();
-    let commit_info_field = commit_info_empty_struct_schema
+    // Remove the inCommitTimestamp field from the schema
+    let commit_info_field = commit_info_schema
         .fields
         .get_mut(COMMIT_INFO_NAME)
         .ok_or_else(|| Error::missing_column(COMMIT_INFO_NAME))?;
-    let DataType::Struct(mut commit_info_data_type) = commit_info_field.data_type().clone() else {
-        return Err(Error::internal_error(
-            "commit_info_field should be a struct",
-        ));
-    };
-    let engine_commit_info_schema =
-        commit_info_data_type.project_as_struct(&["engineCommitInfo"])?;
-    let hack_data_type = DataType::Struct(Box::new(StructType::new(vec![StructField::nullable(
-        "hack_operation_parameter_int",
-        DataType::INTEGER,
-    )])));
-
-    commit_info_data_type
-        .fields
-        .get_mut("operationParameters")
-        .ok_or_else(|| Error::missing_column("operationParameters"))?
-        .data_type = hack_data_type;
-
-    // Since writing in-commit timestamps is not supported, we remove the field so it is not
-    // written to the log
-    commit_info_data_type
-        .fields
-        .shift_remove("inCommitTimestamp");
-    commit_info_field.data_type = DataType::Struct(commit_info_data_type);
 
-    let commit_info_evaluator = engine.evaluation_handler().new_expression_evaluator(
-        engine_commit_info_schema.into(),
-        commit_info_expr,
-        commit_info_empty_struct_schema.into(),
-    );
+    let DataType::Struct(mut commit_info_struct) = commit_info_field.data_type().clone() else {
+        return Err(Error::internal_error("commitInfo field should be a struct"));
+    };
+    commit_info_struct.fields.shift_remove("inCommitTimestamp");
```

Comment:
can we copy over comment why we remove this? and we can take a follow up to properly support ICT :)